src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -535,15 +532,20 @@ private synchronized UUID prepareForReplacement() throws ConfigurationException
             throw new RuntimeException(String.format("Cannot replace_address %s because it doesn't exist in gossip", replaceAddress));
 
         validateEndpointSnitch(epStates.values().iterator());
+        return replaceNodeAndOwnTokens(replaceAddress, epStates, state);
+    }
 
+    @VisibleForTesting
+    public UUID replaceNodeAndOwnTokens(InetAddressAndPort replaceAddress, Map<InetAddressAndPort, EndpointState> epStates, EndpointState state)
```

Comment:
this part of code was synchronized before. We could add synchronization to this method? Also, this probably could have package access.

Reply:
Aha good catch, I missed that :facepalm: The `synchronized` modifier is interesting because I can't see why it's needed. This method is only called from `initServer` -> `prepareToJoin` so I think we should be fine to leave it off?